src/test/java/edu/hm/hafner/analysis/ast/CheckStyleAstFactoryTest.java

```diff
@@ -0,0 +1,795 @@
+    private static final String FINAL_CLASS = "FinalClass";
+    private static final String INTERFACE_IS_TYPE = "InterfaceIsType";
+    private static final String EXPLICIT_INITIALIZATION = "ExplicitInitialization";
+    private static final String PACKAGE_NAME = "PackageName";
+    private static final String NEED_BRACES = "NeedBraces";
+    private static final String METHOD_NAME = "MethodName";
+
+    private static final String METHOD_AST_FOLDERNAME = "MethodAst";
+    private static final String ENVIRONMENT_AST_FOLDERNAME = "EnvironmentAst";
+    private static final String FILE_AST_FOLDERNAME = "FileAst";
+    private static final String CLASS_AST_FOLDERNAME = "ClassAst";
+    private static final String METHOD_OR_CLASS_AST_FOLDERNAME = "MethodOrClassAst";
+    private static final String INSTANCEVARIABLE_AST_FOLDERNAME = "InstancevariableAst";
+    private static final String NAME_PACKAGE_AST_FOLDERNAME = "NamePackageAst";
+
+    // Compare Refactorings from Martin Fowler (http://refactoring.com/catalog/)
+    private static final String REFACTORING_NEWLINE = "_Newline";
+    private static final String REFACTORING_RENAME = "_Rename";
+    private static final String REFACTORING_EXTRACT_METHOD = "_ExtractMethod";
+    private static final String REFACTORING_INLINE_METHOD = "_InlineMethod";
+    private static final String REFACTORING_PULL_UP_METHOD = "_PullUpMethod";
+    private static final String REFACTORING_PUSH_DOWN_METHOD = "_PushDownMethod";
+    private static final String REFACTORING_EXTRACT_CONSTANT = "_ExtractConstant";
+
+    /**
+     * FIXME: In the end this should report no new warnings...
+     */
+    @Test
+    @Disabled("Identifies the problematic ast creation...")
+    void shouldIdentifyTabWarnings() {
+        String file = "Cobertura";
+        String affectedClass = "MavenCoberturaPublisher";
+
+        Set<Issue> affectedPrevious = parse("before/", file, affectedClass);
+        assertEquals("Wrong Number of annotations found: ", 329, affectedPrevious.size());
+
+        Set<Issue> affectedCurrent = parse("after/", file, affectedClass);
+        assertEquals("Wrong Number of annotations found: ", 331, affectedCurrent.size());
+
+        affectedCurrent.removeAll(affectedPrevious);
+        assertEquals("Wrong Number of annotations found: ", 123, affectedCurrent.size());
+
+        Set<String> hashes = new HashSet<String>();
+        HashMap<String, Issue> map = new HashMap<String, Issue>();
+        for (Issue warning : affectedCurrent) {
+            Ast ast = getAst(file + ".java", warning, "", false);
+            String key = ast.getDigest() + warning.getType();
+            if (!hashes.add(key)) {
+                System.out.println("-----------------------");
+                System.out.println("Old: " + map.get(key));
+                System.out.println("-----------------------");
+                System.out.println("Now: " + warning);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [System.out.println is used](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=5019963)